### PR TITLE
Fix the Github Action Recipes for tests, docs, frontend and lint

### LINKED ISSUE #6986: [FEATURE] Implement Github Actions for DjangoCMS
## Feature request 

DjangoCMS has an extensive test suite that runs on Travis. We need to port it to Github actions owing to the issues with super slow builds from Travis. 

Here are some of the links that we can use to help the porting.

- https://developer.okta.com/blog/2020/05/18/travis-ci-to-github-actions
- https://docs.github.com/en/actions/guides/building-and-testing-python

### PR DESCRIPTION
**Pull request by vinitkumar**
## Description

Based on PR from @macolo https://github.com/macolo/django-cms/pull/1. Couldn't create a PR on his fork so creating a PR here. 

## Related resources

- Resolves #6986 

## Checklist

* [ ] I have opened this pull request against ``develop``
* [ ] I have updated the **CHANGELOG.rst**
* [ ] I have added or modified the tests when changing logic

### CODE REVIEW

**Review by macolo — APPROVED** [review]

File: .github/workflows/linters.yml
Comment (Aiky30) [inline .github/workflows/linters.yml:5]: I'm sure that we had isort for the packages too.

**Review by Aiky30 — CHANGES_REQUESTED** [review]
Some repeated comments and clean up. Looks good and you've put a lot of effort in, thanks for that.

File: package.json
Comment (vinitkumar) [reply package.json:25]: This was required because the node-sass packages was not building but then I was testing with latest node.js. Will revert this changes and see if this still builds.

File: .github/workflows/linters.yml
Comment (Aiky30) [reply .github/workflows/linters.yml:5]: Agreed that isort will be tackled separately, right now it's more important to unblock the fact that no tests are running at all in Travis.

**Review by Aiky30 — CHANGES_REQUESTED** [review]
I'm not sure why the suggestions were marked as resolved but not implemented.

File: .github/workflows/test.yml
Comment (Aiky30) [reply .github/workflows/test.yml:94]: This was marked as resolved but it hasn't

File: .github/workflows/test.yml
Comment (Aiky30) [reply .github/workflows/test.yml:48]: This was marked as resolved but it hasn't

**Review by Aiky30 — APPROVED** [review]
Looks good to me. Thanks for addressing my concerns.

**Comment by Aiky30** [comment]
As discussed ideally we would remove the travis-ci.yml file and add a changelog. Not blocking but a nice to have if we can. :-)

**Comment by vinitkumar** [comment]
> As discussed ideally we would remove the travis-ci.yml file and add a changelog. Not blocking but a nice to have if we can. :-)

Just did that in my latest commits. Should be all good now. If I missed anything, let me know.

**Comment by Aiky30** [comment]
We still have the additional tests for the Auth User models, as we are currently blocked this can be incremental and added later. :-)

**Comment by Aiky30** [comment]
@macolo Can you please give this an all clear due to the recent changes made after your approval.

**Comment by Aiky30** [comment]
also we can see the tests are passing in the source branch of the PR because the actions are merged there. Tests pass although this PR is trying to use travis. Github actions will just work.

**Comment by macolo** [comment]
great, looks good to me! Everything passes. Great work!

**Comment by fp4code** [comment]
Hi, is "mysqlclient==2.0.3" necessary?
It is problematic when I do local tests on my computer.
By the way, why no need of psycog2? Either none or both I think (then none I hope).

**Pull request merged**
